bin.src/genDefectList.py

```diff
@@ -0,0 +1,265 @@
+parser.add_argument("--nRows", help="Number of rows in images", type=int, default=4174)
+parser.add_argument("--outFile", help="Output defect list relative to OBS_PFS_DIR", type=str, default="pfs/defects/2015-12-01/defects.dat")
+parser.add_argument("--rerun", help="Which rerun directory are the postISRCCD images in if not in root?", type=str, default="")
+parser.add_argument("--root", help="Path to PFS data directory", type=str, default="/Volumes/My Passport/Users/azuri/spectra/pfs/PFS")
+parser.add_argument("--spectrograph", help="Spectrograph number for which to create the defect list", type=int, default=1, choices=[1, 2, 3, 4])
+parser.add_argument("--verbose", help="Set to print info outputs", action="store_true")
+parser.add_argument("--visitLow", help="Lowest visit number to search for flats", type=int, default=6301)
+parser.add_argument("--visitHigh", help="Highest visit number to search for flats", type=int, default=6758)
+args = parser.parse_args()
+
+#check arguments for validity
+if args.arm not in ['b', 'r', 'm', 'n']:
+    raise RuntimeError("arm not valid")
+if args.expTimeLow <= 0:
+    raise RuntimeError("expTimeLow not valid")
+if args.expTimeHigh <= 0:
+    raise RuntimeError("expTimeHigh not valid")
+if args.expTimeLow == args.expTimeHigh:
+    raise RuntimeError("expTimeLow  and expTimeHigh must be different")
+if args.nCols < 0:
+    raise RuntimeError("nCols not valid")
+if args.nPixCut < 0:
+    raise RuntimeError("nPixCut not valid")
+if args.nRows < 0:
+    raise RuntimeError("nRows not valid")
+if args.gapLow < 0 or args.gapLow >= args.nCols:
+    raise RuntimeError("gapLow not valid")
+if args.gapHigh < 0 or args.gapHigh >= args.nCols or args.gapHigh < args.gapLow:
+    raise RuntimeError("gapHigh not valid")
+if args.spectrograph not in np.arange(1,5,1):
+    raise RuntimeError("spectrograph not in range [1..4]")
+if args.visitLow < 0 or args.visitLow > args.visitHigh:
+    raise RuntimeError("visitLow or visitHigh not valid")
+
+if afwDisplay:
+    if os.environ.get('DISPLAY_DS9_DIR') != "":
+        try:
+            afwDisplay.setDefaultBackend("ds9" if args.display else "virtualDevice")
+        except RuntimeError as e:
+            print e
+
+    afwDisplay.setDefaultMaskTransparency(75)
+
+# Quiet down lots of loggers, so we can see genDefectList logs better
+Log.getLogger("daf.persistence.butler").setLevel(Log.WARN)
+Log.getLogger("afw.image.Mask").setLevel(Log.WARN)
+Log.getLogger("afw.ExposureFormatter").setLevel(Log.WARN)
+Log.getLogger("daf.persistence.LogicalLocation").setLevel(Log.WARN)
+Log.getLogger("CameraMapper").setLevel(Log.WARN)
+Log.getLogger("afw.image.ExposureInfo").setLevel(Log.WARN)
+
+logger = Log.getLogger("genDefectList")
+logger.setLevel(Log.INFO if args.verbose else Log.WARN)
+if args.debug:
+    logger.setLevel(Log.DEBUG)
+
+outFileName = os.path.join(os.environ.get('OBS_PFS_DIR'),args.outFile)
+
+# Read existing defects file
```

Comment:
Added doxygen blocks, reading defects for other detectors using `butler.get('defects',...)`